controllers/operatorpolicy_controller.go

```diff
@@ -108,167 +101,300 @@ func (r *OperatorPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 
 		return reconcile.Result{}, err
 	}
 
 	defer func() {
 		err := r.DynamicWatcher.EndQueryBatch(watcher)
 		if err != nil {
-			OpLog.Error(err, "Could not end query batch for the watcher", "watcher", watcher.Name,
-				"watcherKind", watcher.Kind)
+			OpLog.Error(err, "Could not end query batch for the watcher")
 		}
 	}()
 
 	// handle the policy
-	OpLog.Info("Reconciling OperatorPolicy", "policy", policy.Name)
+	OpLog.Info("Reconciling OperatorPolicy")
 
-	// Check if specified Subscription exists
-	cachedSubscription, err := r.DynamicWatcher.Get(watcher, subscriptionGVK, policy.Spec.Subscription.Namespace,
-		policy.Spec.Subscription.SubscriptionSpec.Package)
-	if err != nil {
-		OpLog.Error(err, "Could not get subscription", "kind", subscriptionGVK.Kind,
-			"name", policy.Spec.Subscription.SubscriptionSpec.Package,
-			"namespace", policy.Spec.Subscription.Namespace)
+	if err := r.handleOpGroup(ctx, policy); err != nil {
+		OpLog.Error(err, "Error handling OperatorGroup")
 
 		return reconcile.Result{}, err
 	}
 
-	subExists := cachedSubscription != nil
+	if err := r.handleSubscription(ctx, policy); err != nil {
+		OpLog.Error(err, "Error handling Subscription")
 
-	// Check if any OperatorGroups exist in the namespace
-	ogNamespace := policy.Spec.Subscription.Namespace
-	if policy.Spec.OperatorGroup != nil {
-		ogNamespace = policy.Spec.OperatorGroup.Namespace
+		return reconcile.Result{}, err
 	}
 
-	cachedOperatorGroups, err := r.DynamicWatcher.List(watcher, operatorGroupGVK, ogNamespace, nil)
+	// FUTURE: more resource checks
+
+	return reconcile.Result{}, nil
+}
+
+func (r *OperatorPolicyReconciler) handleOpGroup(ctx context.Context, policy *policyv1beta1.OperatorPolicy) error {
+	watcher := opPolIdentifier(policy.Namespace, policy.Name)
+
+	desiredOpGroup, err := buildOperatorGroup(policy)
 	if err != nil {
-		OpLog.Error(err, "Could not list operator group", "kind", operatorGroupGVK.Kind,
-			"namespace", ogNamespace)
+		return fmt.Errorf("error building operator group: %w", err)
+	}
 
-		return reconcile.Result{}, err
+	foundOpGroups, err := r.DynamicWatcher.List(
+		watcher, operatorGroupGVK, desiredOpGroup.Namespace, labels.Everything())
+	if err != nil {
+		return fmt.Errorf("error listing OperatorGroups: %w", err)
 	}
 
-	ogExists := len(cachedOperatorGroups) != 0
+	switch len(foundOpGroups) {
+	case 0:
+		// Missing OperatorGroup: report NonCompliance
+		err := r.updateStatus(ctx, policy, missingWantedCond("OperatorGroup"), missingWantedObj(desiredOpGroup))
+		if err != nil {
+			return fmt.Errorf("error updating the status for a missing OperatorGroup: %w", err)
+		}
 
-	// Exists indicates if a Subscription or Operatorgroup need to be created
-	exists := subExists && ogExists
-	shouldExist := strings.EqualFold(string(policy.Spec.ComplianceType), string(policyv1.MustHave))
+		if policy.Spec.RemediationAction.IsEnforce() {
+			err = r.Create(ctx, desiredOpGroup)
```

Comment:
@JustinKuli the oldest version we need to support is OCP 4.12 (https://access.redhat.com/articles/7027073). This seems to map to OLM API v0.17.3:
https://github.com/openshift/operator-framework-olm/blob/release-4.12/go.mod#L16

If you go to this link to compare v0.17.3 to the latest version, you can see that a couple of minor fields were added to the subscription:
https://github.com/operator-framework/api/compare/v0.17.3...v0.21.0

It's easiest to check the crds/operators.coreos.com_subscriptions.yaml file.

So this would mean that using the latest github.com/operator-framework/api is fine but you need to error if those new fields were provided on OCP 4.12.